README.md

```diff
@@ -45,3 +45,81 @@ defmodule YourPlugBasedThing.Endpoint do
   ...
 end
 ```
+
+## Handling Non-standard Path Parameters
+
+Sometimes we have paths that have path parameters that are not easily
+matchable using regular expressions.
+
+A simple example would be this:
+`/api/v1/status_service/slug/my-custom-slug-4`
+
+A more complex example might look like this:
+`/api/v1/status_service/slug/my-custom-slug-4/views/my-user-readable-view-id`
+
+`PlugDatadogStats` will interpret each of these as a separate path and create a tag
+for all of them which is Not Desirable™.
+
+### Some Potential Approaches
+
+#### Anonymous function
+
+This requires the most work from the user of the plug.  However I think this should likely
+be the approach we take until we hear that many people are running in to this issue.
+
+```elixir
+# The function should return a list in the shape of
+# conn.path_info but with fields removed for dd tags
+# Example: ["api", "v1", "status_service", "slug", "OBFUSCATED"]
+
+pre_plug PlugDatadogStats, path_normalize_fn: fn
```

Comment:
Probably worthwhile allowing the user to specify a literal function or a `{Module, Function, Argument}` tuple (arguments would be appended to the path info). Users can still do whatever they want with the incoming path info, but now they can easily store the function wherever they want, with trivial re-use.

Reply:
That's a great suggestion.  Will do!